source/state_representation/README.md

```diff
@@ -236,3 +240,88 @@ StateRepresentation::JointVelocities wVa("a", Eigen::Vector3d(1, 0, 0));
 
 StateRepresentation::JointPositions jp = period * jv; // note that jv * period is also implemented
 ```
+
+## The Jacobian matrix
+
+The `Jacobian` matrix of a robot ensures the conversion between both `CartesianState` and `JointState`.
+Similarly to the `JointState`, a `Jacobian` is associated to a robot and defined by the robot and the number of joints.
+
+```cpp
+// create a jacobian for myrobot with 3 joints
+StateRepresentation::Jacobian jac("myrobot", std::vector<string>({"joint0", "joint1", "joint2"}));
+```
+
+The API is the same as the `JointState`, hence the constructor can also accept the number of joints to initialize the
+joint names vector.
+
+```cpp
+// create a jacobian for myrobot with 3 joints named {"joint0", "joint1", "joint3"}
+StateRepresentation::Jacobian jac("myrobot", 3);
+```
+
+The `Jacobian` is simply a `6 x N` matrix where `N` is the number of joints.
+Therefore, the data can be set from an `Eigen::MatrixXd` of correct dimensions.
+
+```cpp
+jac.set_data(Eigen::MatrixXd::Random(6, 3)); // throw an IncompatibleSizeException if the size is not correct
+```
+
+All the functionalities of the `Jacobian` have been implemented such as `transpose`, `inverse` or `pseudoinverse` functions.
+
+```cpp
+/// returns the 3 x 6 transposed matrix
+StateRepresentation::Jacobian jacT = jac.transpose();
+// will throw an error as a 6 x 3 matrix is not invertible
+StateRepresentation::Jacobian jacInv = jac.inverse();
+// compute the pseudoinverse without the need of being invertible
+StateRepresentation::Jacobian jacPinv = jac.pseudoinverse();
+```
+
+Those operations are very useful to convert `JointState` from `CartesianState` and vice versa.
+
+### Conversion between JointVelocities and CartesianTwist
+
+The simplest conversion is to transform a `JointVelocities` into a `CartesiantTwist` by multiplication with the `Jacobian`
+
+```cpp
+StateRepresentation::JointVelocities jv("myrobot", 3);
+StateRepresentation::CartesianTwist eef_twist = jac * js;
```

Comment:
At some point, we'll need to implement a check that the multiplication is compatible in terms of names and reference names right? It will go with the jacobian refactoring? I'm not even sure anymore because all these operators and operations mess up my brain slowly but steadily :smile:

Reply:
compatibility with `JointState` is already implemented. That is actually what motivated #43 as I was having an error on the iiwa. For `CartesianState` yes it will have to go with the refactor.